docs/source/deploying-volttron/proxy_web_server.rst

```diff
@@ -0,0 +1,323 @@
+        DocumentRoot "<VOLTTRON_ROOT>/volttron/volttron/platform/web/static"
+
+        # Set Directory file path to match DocumentRoot
+        <Directory "<VOLTTRON_ROOT>/volttron/volttron/platform/web/static">
+            Options Indexes FollowSymLinks
+            AllowOverride None
+            Require all granted
+        </Directory>
+
+        # Force redirect of http to https
+        RewriteEngine On
+        RewriteCond %{HTTPS} off
+        RewriteRule ^/+(.*) https://%{HTTP_HOST}%{REQUEST_URI} [R=permanent,L]
+    </VirtualHost>
+
+
+ssl.conf File
+-------------
+
+The ssl.conf contains the proxy, cert, and https configuration details. By default, the file can be found
+at /etc/apache2/mods-enabled/ssl.conf.
+
+An example ssl.conf file is provided with volttron and can be found at volttron/scripts/admin/apache-proxy/ssl.conf.
+The file will need to be modified. The SSLCertificateFile, and SSLCertificateKeyFile paths need to be set to appropriate
+cert and key file path.
+
+Generating self-signed Certs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following instructions are based on directions for generating a self-signed certificate
+provided by Ubuntu: `https://ubuntu.com/server/docs/security-certificates <https://ubuntu.com/server/docs/security-certificates>`_.
+
+
+.. warning::
+
+    While we generate a self-signed certificate for the demo, it is not recommended to use a self-signed certificate
+    in a production environment, as this is not a secure practice. Instead use your institution's CA and certificates or a
+    trusted 3rd party, e.g. `Let's Encrypt <https://letsencrypt.org/>`_. This is provided for education and demonstrative
+    purposes only.
+
+First, generate a server key:
+
+.. code-block::
+
+    openssl genrsa -des3 -out server.key 2048
+
+This key will require a pass-phrase on startup of the proxy service which can become inconvenient.
+It is possible to create a version of the key that does not require a pass-phrase, though this is less secure.
+This will be done for this demo, but is not recommended (see warning and link above)
+
+.. code-block::
+    openssl rsa -in server.key -out server.key.insecure
+    mv server.key server.key.secure
+    mv server.key.insecure server.key
+
+Generate a CSR.
+
+.. code-block::
+    openssl req -new -key server.key -out server.csr
+
+Create the certificate file using the CSR and key files:
+
+.. code-block::
+
+    openssl x509 -req -days 365 -in server.csr -signkey server.key -out server.crt
+
+server.key can now be used for the SSLCertificateKeyFile entry,
+and server.crt can now be used for the SSLCertificateFile entry.
+
+
+ Once that is complete, verify that ProxyPass and ProxyPassReverse options,
+as well as the /vc path found under the RewriteRule option are all appropriate for your use case. This simple use-case
+routes all web traffic through the proxy to your VOLTTRON instance running on your localhost.
+
+
+.. code-block:: console
+
```

Comment:
which file is this?   Add # ssl.conf at the top?

Reply:
Added. Along with the other two files.